### ISSUE TITLE
If Statement Very Slow

### ISSUE THREAD

**Issue by tonychenxyz** [pytorch/xla#2855]
## 🐛 Bug

if statement seems to be very slow when using PyTorch on TPU compared to that on GPU.

Here's what happens on GPU:
![image](https://user-images.githubusercontent.com/13805929/113381116-b4190d80-93b0-11eb-86de-085e91e7a3f5.png)

Here's what happens on TPU:
![image](https://user-images.githubusercontent.com/13805929/113381149-d0b54580-93b0-11eb-94e2-71df617c490a.png)

tokenizer.stoi["<eos>"] is a simple integer.

**Comment by JackCaoG** [comment]
I tried
```
t1 = torch.tensor(1, device=xla_device)                                                                           
t2 = torch.randn(10, 10, device=xla_device)

start = time.time()                                                                                               
for i in range(1000):                                                                                             
    torch.argmax(t2) == t1                                                                                        
end = time.time()   
```

and use 
```
import torch_xla.debug.metrics as met
print(met.metrics_report()) 
```
to print the metric. Above code actually did not get executed on TPU so it is "fast".  If I put above code into a if statement I see
```
Metric: CompileTime
  TotalSamples: 1
  Accumulator: 206ms542.491us
  Percentiles: 1%=206ms542.491us; 5%=206ms542.491us; 10%=206ms542.491us; 20%=206ms542.491us; 50%=206ms542.491us; 80%=206ms542.491us; 90%=206ms542.491us; 95%=206ms542.491us; 99%=206ms542.491us
Metric: ExecuteTime
  TotalSamples: 1000
  Accumulator: 02s926ms168.141us
  ValueRate: 539ms674.276us / second
  Rate: 279.661 / second
  Percentiles: 1%=001ms291.910us; 5%=001ms380.797us; 10%=001ms447.096us; 20%=002ms523.284us; 50%=002ms741.162us; 80%=002ms990.834us; 90%=002ms133.016us; 95%=002ms271.932us; 99%=003ms523.371us
```


PT/XLA tensor is lazy and only get executed when  the output of the tensor is required. My guess is that python code bypass the `==` since it does not affect the result.


On a separate note, calling `.cpu()` will result in a data transfer from TPU to user vm via networking under the current 2vm setup and it is expected to be slower. We generally don't recommend inspect the value of the tensor on device before marking a step. Doing this benchmark is not very useful. To get the best performance out of pt/xla with TPU, model only get execute once at the end of an epoch instead of 1000 micro executions in the above example.

More detail in https://github.com/pytorch/xla/blob/master/TROUBLESHOOTING.md#known-performance-caveats
